src/Elastic.Apm/Model/Transaction.cs

```diff
@@ -132,10 +132,16 @@ internal Transaction(ApmAgent agent, string name, string type, long? timestamp =
 			var spanLinks = links as SpanLink[] ?? links?.ToArray();
 			Links = spanLinks;
 
+			// Restart the trace when:
+			// - `TraceContinuationStrategy == Restart` OR
+			// - `TraceContinuationStrategy == RestartExternal` AND
+			//		- `TraceState` is not present (Elastic Agent would have added it) OR
+			//		- `TraceState` is present but the SampleRate is not present (Elastic agent adds SampleRate to TraceState)
 			var shouldRestartTrace = configuration.TraceContinuationStrategy == ConfigConsts.SupportedValues.Restart ||
-				configuration.TraceContinuationStrategy == ConfigConsts.SupportedValues.RestartExternal
-				&& distributedTracingData is { TraceState: { SampleRate: null } };
+				(configuration.TraceContinuationStrategy == ConfigConsts.SupportedValues.RestartExternal
+					&& (distributedTracingData?.TraceState == null || distributedTracingData is { TraceState: { SampleRate: null } }));
 
+			// For each new transaction, start an Activity if we're not ignoring them.
```

Comment:
This comment was duplicated from the line below.

Reply:
Hahh, thanks! Fixed